packages/app/src/cli/prompts/dev.test.ts

```diff
@@ -100,25 +102,21 @@ describe('selectApp', () => {
   it('returns app if user selects one', async () => {
     // Given
     const apps: OrganizationApp[] = [APP1, APP2]
-    vi.mocked(ui.prompt).mockResolvedValue({apiKey: 'key2'})
+    vi.mocked(renderAutocompletePrompt).mockResolvedValue('key2')
 
     // When
     const got = await selectAppPrompt(apps, ORG1.id, 'token')
 
     // Then
-    expect(got).toEqual({apiKey: APP2.apiKey})
-    expect(ui.prompt).toHaveBeenCalledWith([
-      {
-        type: 'autocomplete',
-        name: 'apiKey',
-        message: 'Which existing app is this for?',
-        choices: [
-          {name: 'app1', value: 'key1'},
-          {name: 'app2', value: 'key2'},
-        ],
-        source: expect.any(Function),
-      },
-    ])
+    expect(got).toEqual(APP2.apiKey)
+    expect(renderAutocompletePrompt).toHaveBeenCalledWith({
```

Comment:
I totally agree it would be better, but in this case because the autocomplete prompt can change its appearance based on its state we would need some extra logic to get the first frame only.

I'm going to tackle testing outside of components a second PR. For now, because we have good component unit tests we can be sure that if certain params are passed in then the output is what we would expect.